queries/rust/locals.scm

```diff
@@ -0,0 +1,100 @@
+; Imports
+(extern_crate_declaration
+    name: (identifier) @definition.import)
+
+(use_declaration 
+  argument: (scoped_identifier 
+              name: (identifier) @definition.import))  
+
+(use_as_clause 
+  alias: (identifier) @definition.import)
+
+(use_list
+    (identifier) @definition.import) ; use std::process::{Child, Command, Stdio};
+
+; Functions
+(function_item 
+    name: (identifier) @definition.function)
+
+(function_item 
+  name: (identifier) @definition.method
+  parameters: (parameters 
+                (self_parameter)))
+
+; Variables
+(parameter 
+  pattern: (identifier) @definition.var) 
+
+(let_declaration 
+  pattern: (identifier) @definition.var)
+
+(const_item 
+  name: (identifier) @definition.var)
+
+(tuple_pattern
+  (identifier) @definition.var) 
+
+(if_let_expression
+  pattern: (_
+             (identifier) @definition.var))
+
+(tuple_struct_pattern
+  (identifier) @definition.var)
+
+(closure_parameters
+  (identifier) @definition.var)
+
+(self_parameter
+  (self) @definition.var)
+
+(for_expression 
+  pattern: (identifier) @definition.var)
+
+; Types
+(struct_item
+  name: (type_identifier) @definition.type)
+
+(constrained_type_parameter 
+  left: (type_identifier) @definition.type) ; the P in  remove_file<P: AsRef<Path>>(path: P)
+
+(enum_item
+  name: (type_identifier) @definition.type)
+
+
+; Fields
+(field_declaration
+  name: (field_identifier) @definition.field)
+
+(enum_variant
+  name: (identifier) @definition.field) 
+
+; References
+(identifier) @reference
+(type_identifier) @reference
+(field_identifier) @reference
```

Comment:
Same issue as in #164. I think we should also consider references to types and fields.

Reply:
I agree on this in the end

Reply:
Should it be `@reference.type`?

Reply:
Yes, I like that more !